integration-tests/tests/src/tests/sync/sync-session.ts

```diff
@@ -286,16 +288,14 @@ describe("SessionTest", () => {
   describe("progress notification", () => {
     afterEach(() => Realm.clearTestState());
     it("is called", async function (this: AppContext) {
+      this.timeout(5000);
       const partition = generatePartition();
       const { config } = await getSyncConfWithUser(this.app, partition);
-      let progressCalled = false;
-      await Promise.race([
-        Realm.open(config).progress(() => {
-          progressCalled = true;
-        }),
-        throwAfterTimeout(5000),
-      ]);
-      expect(progressCalled).to.be.true;
+
+      const progressCallback = spy();
+      await Realm.open(config).progress(progressCallback);
+
+      expect(progressCallback.called).is.true;
```

Comment:
Not going to spend much time refactoring the other tests since they related to the deprecated method, just thought this was a nice opportunity for sinon (also just timeout).

Reply:
Also, there's a sinon-chai integration and I will switch to that soon, maybe in a different PR if there's not enough time for now